### PR TITLE
NDEV-3277. Fix different number of accounts in debug_traceTransaction

### PR DESCRIPTION
**Pull request by ancientmage**

### CODE REVIEW

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/10704713613/artifacts/1891958635)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getSlot                           |   61100 |      0     |      0.169 |    0.000956874 |        0      |
| getAccountInfo                    |   59180 |      0     |      0.105 |    0.00130612  |        0      |
| getBlocks                         |   24307 |      0     |      0.026 |    0.000591023 |        0      |
| getBlock                          |   15724 |      0     |      0.033 |    0.00128975  |        0.001  |
| getMultipleAccounts               |   14266 |      0     |      0.108 |    0.00207374  |        0.001  |
| getLatestBlockhash                |   11286 |      0     |      0.097 |    0.00197643  |        0      |
| getTransaction                    |    8142 |      0     |      0.111 |    0.0104587   |        0.005  |
| getBlockTime                      |    6400 |      0     |      0.035 |    0.000392031 |        0      |
| sendTransaction                   |    5285 |      0     |      0.104 |    0.00872091  |        0.004  |
| getBalance                        |    1845 |      0     |      0.04  |    0.00141897  |        0.001  |
| getSignatureStatuses              |    1186 |      0     |      0.085 |    0.00110371  |        0      |
| getVersion                        |     625 |      0     |      0.033 |    0.001312    |        0      |
| isBlockhashValid                  |     575 |      0     |      0.021 |    0.000431304 |        0      |
| getSignaturesForAddress           |      61 |      0     |      0.158 |    0.00372131  |        0.001  |
| simulateTransaction               |      38 |      0     |      0.001 |    0.000447368 |        0      |
| requestAirdrop                    |      32 |      0     |      0.001 |    0.0006875   |        0.001  |
| getTokenAccountBalance            |       7 |      0     |      0.001 |    0.000285714 |        0      |
| getHealth                         |       3 |      0     |      0.001 |    0.000666667 |        0.001  |
| getFirstAvailableBlock            |       2 |      0     |      0.001 |    0.0005      |        0.0005 |
| getMinimumBalanceForRentExemption |       1 |      0     |      0     |    0           |        0      |
| getClusterNodes                   |       1 |      0.002 |      0.002 |    0.002       |        0.002  |

</details>

File: evm_loader/lib/src/commands/emulate.rs [hunk 1]
```diff
@@ -252,22 +249,31 @@ async fn emulate_trx<T: Tracer>(
...
+        accounts_pubkey.push(solana_account.pubkey);
+    }
+
+    let result = storage
+        .get_multiple_accounts(accounts_pubkey.as_slice())
+        .await?;
+
+    for (pubkey, account) in accounts_pubkey.iter_mut().zip(result.iter()) {
+        if let Some(account) = account {
+            accounts_data.push(AccountData::new_from_account(*pubkey, account));
```
Comment (s-medvedev) [inline evm_loader/lib/src/commands/emulate.rs]: FYI Rust-way:
```rust
    let pubkeys = solana_accounts.iter().filter_map(|v|
        if v.is_writable || AccountInfoLevel::Changed != *level {
            Some(v.pubkey)
        } else {
            None
        }
    ).collect::<Vec<_>>();

    let result = storage
        .get_multiple_accounts(&pubkeys)
        .await?;

    let accounts_data = pubkeys.iter().zip(result.into_iter())
        .filter_map(|(pubkey, account)| {
            account.map(|acc| AccountData::new_from_account(*pubkey, &acc))
        })
        .collect::<Vec<_>>();
```

**Review by s-medvedev — DISMISSED** [review]

File: evm_loader/lib/src/commands/emulate.rs (hunk 1 above)
Comment (ancientmage) [reply evm_loader/lib/src/commands/emulate.rs]: fixed

Comment (ancientmage) [reply evm_loader/lib/src/commands/emulate.rs]: fixed

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/10736101603/artifacts/1900574802)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getSlot                           |   66395 |      0     |      0.189 |    0.00132713  |         0.001 |
| getAccountInfo                    |   59757 |      0     |      0.166 |    0.00256825  |         0.001 |
| getBlocks                         |   27271 |      0     |      0.153 |    0.00094441  |         0.001 |
| getBlock                          |   17831 |      0     |      0.229 |    0.00234104  |         0.002 |
| getMultipleAccounts               |   14850 |      0     |      0.181 |    0.00294088  |         0.001 |
| getLatestBlockhash                |   12219 |      0     |      0.164 |    0.00205156  |         0.001 |
| getTransaction                    |    8186 |      0.001 |      0.183 |    0.0146719   |         0.005 |
| getBlockTime                      |    6342 |      0     |      0.107 |    0.00107632  |         0     |
| sendTransaction                   |    5321 |      0     |      0.207 |    0.010412    |         0.005 |
| getBalance                        |    1861 |      0     |      0.088 |    0.00298442  |         0.001 |
| getSignatureStatuses              |    1198 |      0     |      0.084 |    0.00251753  |         0.001 |
| getVersion                        |     638 |      0     |      0.125 |    0.00368339  |         0.001 |
| isBlockhashValid                  |     580 |      0     |      0.027 |    0.000748276 |         0     |
| getSignaturesForAddress           |      63 |      0     |      0.003 |    0.00188889  |         0.002 |
| simulateTransaction               |      38 |      0     |      0.003 |    0.000789474 |         0.001 |
| requestAirdrop                    |      32 |      0.001 |      0.004 |    0.00121875  |         0.001 |
| getTokenAccountBalance            |      10 |      0     |      0.07  |    0.0094      |         0.001 |
| getHealth                         |       3 |      0.001 |      0.002 |    0.00133333  |         0.001 |
| getFirstAvailableBlock            |       2 |      0.001 |      0.001 |    0.001       |         0.001 |
| getMinimumBalanceForRentExemption |       1 |      0     |      0     |    0           |         0     |
| getClusterNodes                   |       1 |      0     |      0     |    0           |         0     |

</details>

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**